lib/phoenix/router.ex

```diff
@@ -357,6 +358,10 @@ defmodule Phoenix.Router do
                  unquote(exprs.host)) do
         unquote(exprs.dispatch)
       end
+      def __match_params__(_conn, unquote(exprs.verb_match), unquote(exprs.path),
+                 unquote(exprs.host)) do
+        unquote(exprs.binding)
+      end
```

Comment:
It feels like we should standardize privately injected apis under the `__phoenix__(..` or `__router__` convention, but the ship may have sailed with our other injections at this point, i.e. `__routes__`. Thoughts?